nexus/db-queries/src/db/datastore/inventory.rs

```diff
@@ -134,6 +135,23 @@ impl DataStore {
                 ))
             })
             .collect::<Result<Vec<_>, Error>>()?;
+
+        // Pull disk firmware out of sled agents
+        let disk_firmware: Vec<_> = collection
+            .sled_agents
+            .iter()
+            .flat_map(|(sled_id, sled_agent)| {
+                sled_agent.disks.iter().map(|disk| {
+                    InvNvmeDiskFirmware::try_from_physical_disk(
```

Comment:
My only beef with this setup is that, when we later add a non-Nvme firmware type, we won't have any indication that this should change. Basically: This would silently drop non-nvme firmware disks.

Any way we could do something like:

```rust
match disk.firmware {
  Nvme => // Convert to Some(InvNvmeDiskFirmware)
  Unknown => // None
}
```

(It actually might make sense to have an enum around the possible DB types -- maybe `InvDiskFirmware` -- like `PhysicalDiskFirmware` does, so we can let `disk_firmware` be a `Vec` of many possible FW types)